mirascope/base/calls.py

```diff
@@ -109,3 +111,40 @@ def _setup(
             tool_types = kwargs.pop("tools")
             kwargs["tools"] = [tool_type.tool_schema() for tool_type in tool_types]
         return kwargs, tool_types
+
+    @classmethod
+    def from_call(
+        cls, base_call_type: type[BaseCallT], call_params: BaseCallParams
+    ) -> type[BaseCallT]:
+        """Dynamically creates a new call class from a base model and a call.
+
+        Args:
+            cls: The call to use for the base model.
+            base_call_type: The base model to use for the call. The new call will
+                inherit all the fields and class variables from this model.
+            call_params: The call params to use for the call.
+
+        Returns:
+            A new call class.
+        """
+
+        fields: dict[str, Any] = {
+            name: (field.annotation, field.default)
+            for name, field in base_call_type.model_fields.items()
+        }
+
+        class_vars = {
+            name: value
+            for name, value in base_call_type.__dict__.items()
+            if name not in base_call_type.model_fields
+        }
+        new_call = create_model(base_call_type.__name__, __base__=cls, **fields)
+
+        for var_name, var_value in class_vars.items():
+            setattr(new_call, var_name, var_value)
+        setattr(new_call, "call_params", call_params)
+
+        return cast(type[BaseCallT], new_call)
```

Comment:
shouldn't we just use `__base__=call_type` in `create_model` and then not cast here?

Reply:
note: this is only correct when updating `base_call_type` to `call_type`